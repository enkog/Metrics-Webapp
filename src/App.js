function App() {
  return (
    <div>
      <h1>Metrics Webapp</h1>
    </div>
  );
}

export default App;
