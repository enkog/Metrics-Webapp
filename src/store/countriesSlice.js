import { createSlice } from '@reduxjs/toolkit';

const countriesSlice = createSlice({
  name: 'countries',
  initialState: { countries: [], loading: false },
  reducers: {},
});

export const countriesActions = countriesSlice.actions;

export default countriesSlice;
