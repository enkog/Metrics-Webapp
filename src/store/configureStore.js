import { configureStore } from '@reduxjs/toolkit';
import countriesSlice from './countriesSlice';

const store = configureStore({
  reducer: { countries: countriesSlice.reducer },
});

export default store;
