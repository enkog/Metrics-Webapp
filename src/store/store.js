import { configureStore } from '@reduxjs/toolkit';
import logger from 'redux-logger';
import thunk from 'redux-thunk';
import countriesSlice from './countriesSlice';
import detailsSlice from './detailsSlice';

const store = configureStore({
  reducer: { countries: countriesSlice.reducer, details: detailsSlice.reducer },
  middleware: [thunk, logger],
});

export default store;
