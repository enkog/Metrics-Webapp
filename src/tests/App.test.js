import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/store';
import App from '../App';

describe('App Component', () => {
  test('renders the App component', () => {
    const component = renderer.create(
      <Provider store={store}>
        <Router>
          <App />
        </Router>
      </Provider>,
    )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
