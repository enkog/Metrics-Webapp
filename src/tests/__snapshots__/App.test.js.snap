// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component renders the App component 1`] = `
<div>
  <nav
    className="navbar navbar-expand navbar-light"
  >
    <a
      aria-current="page"
      className="d-flex align-items-center text-white ps-2 w-100 active"
      href="/"
      onClick={[Function]}
      style={Object {}}
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 256 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z"
        />
      </svg>
       
      Countries
    </a>
    <div
      className="ms-auto navbar-nav"
      onKeyDown={[Function]}
    >
      <input
        className=" input-field mx-1"
        onChange={[Function]}
        placeholder="Search by regions"
        type="text"
      />
    </div>
  </nav>
  <div
    className="countries"
  >
    <div
      className="card-group g-0"
    />
  </div>
</div>
`;
